File: MODULO3/PYTHON/TRABAJANDO_PYTHON/asociacion_muchos_a_muchos_estudiante_curso/estudiante.py
class Estudiante:
    def __init__(self, idEstudiante="",nombre=""):
        self._idEstudiante = idEstudiante
        self._nombre = nombre
        self.inscripciones = []

    @property
    def idEstudiante(self):
        return self._idEstudiante
    
    @idEstudiante.setter
    def idEstudiante(self, value):
        self._idEstudiante = value

    @property
    def nombre(self):
        return self._nombre
    @nombre.setter
    def nombre(self, value):
        self._nombre = value

    def __str__(self):
        return f"Estudiante --> Nombre: {self.nombre}, ID: {self.idEstudiante}"

    def inscribir_curso(self, curso, fecha_inscripcion):
        inscripcion = Inscripcion(self, curso, fecha_inscripcion)
        self.inscripciones.append(inscripcion)
        

File: MODULO3/UML/DiagramaUML/src/asociacion_tipo_muchos_a_muchos_estudiante_curso_mysql/Estudiante.java
package asociacion_tipo_muchos_a_muchos_estudiante_curso_mysql;

import java.util.ArrayList;
import java.util.List;

public class Estudiante {

    // VARIABLES DE INSTANCIA
    private String idEstudiante;
    private String nombre;
    private List<Inscripcion> inscripciones; //Relación con inscripción

    // CONSTRUCTORES
    public Estudiante() {
        this.idEstudiante = "";
        this.nombre = "";
        this.inscripciones = null;
    }

    public Estudiante(String idEstudiante, String nombre) {
        this.idEstudiante = idEstudiante;
        this.nombre = nombre;
        inscripciones = new ArrayList<>();
    }
    
    // METODOS SET Y GET
    public String getIdEstudiante() {
        return idEstudiante;
    }

    public void setIdEstudiante(String idEstudiante) {
        this.idEstudiante = idEstudiante;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }
    
    // METODO TOSTRING
    @Override
    public String toString() {
        return "Estudiante{" + "idEstudiante=" + idEstudiante + ", nombre=" + nombre + '}';
    }
    
    // METODO RELACION INSCRIPCION
    public void inscribirCurso(Curso curso, String fechaInscripcion) {
        Inscripcion inscripcion = new Inscripcion(this,curso,fechaInscripcion);
        this.inscripciones.add(inscripcion);
    }
}
